static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx

```diff
@@ -104,20 +139,43 @@ function SentryFunctionDetails(props: Props) {
             </PanelBody>
           </Panel>
         </Form>
+        {sentryFunction && (
+          <Button
+            onClick={handleDelete}
+            title={t('Delete Sentry Function')}
+            aria-label={t('Delete Sentry Function')}
+            type="button"
+            priority="danger"
+          >
+            Delete Sentry Function
+          </Button>
+        )}
       </Feature>
     </div>
   );
 }
 
-type WrapperState = {} & AsyncComponent['state'];
+type WrapperState = {
+  sentryFunction: SentryFunction;
+} & AsyncComponent['state'];
 
 type WrapperProps = {
   params: {orgId: string; functionSlug?: string};
+  sentryFunction?: SentryFunction;
```

Comment:
@vpandiarajan20 is this needed?

Reply:
I needed a way to pass in the sentryFunction into the functional component, so the easiest way to do that was to get it in the class component and pass it as a prop to the functional component.